docs/runtime/resume.md

```diff
@@ -0,0 +1,241 @@
+information, see 
+[Add resume to custom Agents](#custom-agents).
+
+## Add resumable configuration
+
+Enable the resume function for an agent workflow by applying a Resumabiltiy
+configuration to the App object of your ADK workflow, as shown in the following
+code example:
+
+```python
+app = App(
+    name='human_tool_confirmation',
+    root_agent=root_agent,
+    # Set the resumability config to enable resumability.
+    resumability_config=ResumabilityConfig(
+        is_resumable=True,
+    ),
+)
+```
+
+!!! warning "Caution: Long Running Functions, Confirmations, Authentication"
+    For agents that use
+    [Long Running Functions](/adk-docs/tools/function-tools/#long-run-tool),
+    [Confirmations](/adk-docs/tools/confirmation/), or
+    [Authentication](/adk-docs/tools/authentication/)
+    requiring user input, adding a resumable confirmation changes how these features
+    operate. For more information, see the documentation for those features.
+
+!!! info "Note: Custom Agents"
+    Resume is not supported by default for Custom Agents. You must
+    update the agent code for a Custom Agent to support the Resume feature. For
+    information on modifying Custom Agents to support incremental resume
+    functionality, see 
+    [Add resume to custom Agents](#custom-agents).
+
+## Resume a stopped workflow
+
+When an ADK workflow stops execution you can resume the workflow using a
+command containing the Invocation ID for the workflow instance, which can be
+found in the
+[Event](/adk-docs/events/#understanding-and-using-events)
+history of the workflow. Make sure the ADK API server is running, in case it was
+interrupted or powered off, and then run the following command to resume the
+workflow, as shown in the following API request example.
+
+```console
+# restart the API server if needed:
+adk api_server my_resumable_agent/
+
+# resume the agent:
+curl -X POST http://localhost:8000/run_sse \
+ -H "Content-Type: application/json" \
+ -d '{
+   "app_name": "my_resumable_agent",
+   "user_id": "u_123",
+   "session_id": "s_abc",
+   "invocation_id": "invocation-123",
+ }'
+```
+
+You can also restart a workflow using the Runner object Run Async method, as
+shown below:
+
+```python
+runner.run_async(user_id='u_123', session_id='s_abc', 
+    invocation_id='invocation-123' ,new_message=)
+
+# when new_message is set to a function response,
+# we are trying to resume a long running function.
+```
+
+!!! info "Note"
+    Resuming a workflow from the ADK Web user interface or using the ADK
+    command line (CLI) tool is not currently supported.
+
+## How it works
+
+The resume functionality works by logging completed Agent workflow tasks,
+including incremental steps using
+[Events](/adk-docs/events/) and
+[Event Actions](/adk-docs/events/#detecting-actions-and-side-effects).
+tracking completion of agent tasks within a resumable workflow. If a workflow is
+interrupted and then later restarted, the resume functionality reinstates the
```

Comment:
Should we just say "ADK reinstates the interrupted workflow" rather than "the resume functionality"? 

Same for other paragraphs.